### PR TITLE
libbsd: add libbsd/0.10.0 recipe

### PR DESCRIPTION
**Pull request by madebr**
Specify library name and version:  **libbsd/0.10.0**

@ItsBasi Useful for #2415

Fixes https://github.com/conan-io/wishlist/issues/185

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/libbsd/all/test_package/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,10 @@
+cmake_minimum_required(VERSION 2.8.11)
```
Comment (madebr) [inline recipes/libbsd/all/test_package/CMakeLists.txt:1]:
```suggestion
cmake_minimum_required(VERSION 3.1)
```

**Comment by Croydon** [comment]
Please put `Fixes https://github.com/conan-io/wishlist/issues/185` in the PR, thanks!

**Comment by madebr** [comment]
> Please put `Fixes https://github.com/conan-io/wishlist/issues/185` in the PR, thanks!

Added it.

**Comment by madebr** [comment]
@itsbasi It seems libbsd only builds on Linux, so something to incorporate in your minizip recipe.

**Comment by ItsBasi** [comment]
Had a look at the MacOS build failure. According to this [pull request](https://gitlab.freedesktop.org/libbsd/libbsd/-/merge_requests/3), macos/darwin needs to be patched. The request also notes that nixpkgs has a [patch](https://github.com/NixOS/nixpkgs/blob/1016bfcff1df163aff75d908df1e00f897a29b9b/pkgs/development/libraries/libbsd/darwin.patch) for libbsd, which might be usable.

File: recipes/libbsd/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    }
+
+    _autotools = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
```
Comment (uilianries) [inline recipes/libbsd/all/conanfile.py:32]: not required, but it doesn't harm

File: recipes/libbsd/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+            del self.options.fPIC
+
+    def configure(self):
```
Comment (madebr) [inline recipes/libbsd/all/conanfile.py:35]:
```suggestion
        return "source_subfolder"

    def configure(self):
```

**Comment by ItsBasi** [comment]
Thanks for taking care of libbsd

**Pull request merged**